### PR TITLE
[Leetcode113][Path Sum 2][Xia Zhao][2021-08-04][Please review]

### PR DESCRIPTION
**Pull request by xiazhao9406**

### CODE REVIEW

File: src2/p0113/Solution.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ *     <li><a href="https://www.programmersought.com/article/78473293257/">Programmer Sought</a></li>
+ *   </ul>
+ * <H3>Related Topics:</H3>
+ *   <ul>
+ *     <li>Binary Tree</li>
+ *   </ul>
+ */
+public class Solution {
+    public List<List<Integer>> pathSum(TreeNode root, int sum) {
+        LinkedList<Integer> list = new LinkedList<>();
```
Comment (theodoregao) [inline src2/p0113/Solution.java:20]: Add `final` for immutable variables.
`final LinkedList<Integer> list = new LinkedList<>();`

File: src2/p0113/Solution.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        List<List<Integer>> result = new ArrayList<List<Integer>>();
```
Comment (theodoregao) [inline src2/p0113/Solution.java:21]: Add `final` for immutable variables.
`final List<List<Integer>> result = new ArrayList<List<Integer>>();`

File: src2/p0113/Solution.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        findPath(root,sum,0,list,result);
```
Comment (theodoregao) [inline src2/p0113/Solution.java:22]: Please add space between each parameter.
`findPath(root, sum, 0, list, result);`

File: src2/p0113/Solution.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
+package p0113;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * <H3>Leetcode problem 113</H3><a href="https://leetcode.com/explore/challenge/card/august-leetcoding-challenge-2021/613/week-1-august-1st-august-7th/3838//">Path Sum 2</a>
```
Comment (theodoregao) [inline src2/p0113/Solution.java:8]: Remove extra `/` by the end of url.
`* <H3>Leetcode problem 113</H3><a href="https://leetcode.com/explore/challenge/card/august-leetcoding-challenge-2021/613/week-1-august-1st-august-7th/3838/">Path Sum 2</a>`

File: src2/p0113/Solution.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ */
+public class Solution {
+    public List<List<Integer>> pathSum(TreeNode root, int sum) {
+        LinkedList<Integer> list = new LinkedList<>();
+        List<List<Integer>> result = new ArrayList<List<Integer>>();
+        findPath(root,sum,0,list,result);
+        return result;
+    }
+
+    public void findPath(TreeNode root,int sum,int curSum,LinkedList<Integer> list,List<List<Integer>> result){
```
Comment (theodoregao) [inline src2/p0113/Solution.java:26]: Add spaces between parameters.

File: src2/p0113/Solution.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+ * <H3>References:</H3>
+ *   <ul>
+ *     <li><a href="https://www.programmersought.com/article/78473293257/">Programmer Sought</a></li>
+ *   </ul>
+ * <H3>Related Topics:</H3>
+ *   <ul>
+ *     <li>Binary Tree</li>
+ *   </ul>
+ */
+public class Solution {
```
Comment (theodoregao) [inline src2/p0113/Solution.java:18]: The `Solution` class doesn't need to be a `public` class.

File: src2/p0113/Solution.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    public List<List<Integer>> pathSum(TreeNode root, int sum) {
+        LinkedList<Integer> list = new LinkedList<>();
+        List<List<Integer>> result = new ArrayList<List<Integer>>();
+        findPath(root,sum,0,list,result);
+        return result;
+    }
+
+    public void findPath(TreeNode root,int sum,int curSum,LinkedList<Integer> list,List<List<Integer>> result){
+        if(root == null)
+            return;
```
Comment (theodoregao) [inline src2/p0113/Solution.java:28]: You need a space after `if`, and keep your coding style consistent with `{ }` for each code block.
```
        if (root == null) {
            return;
        }
```

File: src2/p0113/Solution.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        list.add(root.val);
+        curSum += root.val;
+
+        if(root.left == null && root.right == null && sum == curSum){
+            result.add(new LinkedList<>(list));
+        }
+
+        findPath(root.left,sum,curSum,list,result);
+        findPath(root.right,sum,curSum,list,result);
```
Comment (theodoregao) [inline src2/p0113/Solution.java:38]: Add spaces between parameters.

**Review by theodoregao — CHANGES_REQUESTED** [review]

File: src2/p0113/Solution.java (hunk 6 above)
Comment (xiazhao9406) [reply src2/p0113/Solution.java:18]: yes, it needed

**Pull request merged**